File: adventofcode/year2015/day02.py
from adventofcode import Solution


class Day02(Solution):
    def __init__(self, year: str, day: str):
        super().__init__(year, day)

        self._input = [l.strip().split('x') for l in self._load_input_as_lines()]

    def part_one(self):
        return sum([2*l*w + 2*w*h + 2*h*l + min(l*w, w*h, l*h) for (l, w, h) in map(lambda d: (int(d[0]), int(d[1]), int(d[2])), self._input)])

    def part_two(self):
        return sum([l*w*h + min((2*l+2*w), (2*w+2*h), (2*h+2*l)) for (l, w, h) in map(lambda d: (int(d[0]), int(d[1]), int(d[2])), self._input)])

File: adventofcode/year2017/Day02.cs
using adventofcode.common;
using adventofcode.common.util;

namespace adventofcode.year2017;

public class Day02 : Solution
{
    private List<List<int>> _input;
    
    public Day02(string year, string day) : base(year, day)
    {
        _input = new List<List<int>>();
        foreach (var line in LoadInputAsLines())
        {
            _input.Add(line.Split("\t").Select(x => Convert.ToInt32(x)).ToList());
        }
    }

    public override string PartOne()
    {
        var checksum = _input.Aggregate(0, (acc, row) => acc + row.Max() - row.Min());
        return Convert.ToString(checksum);
    }

    public override string PartTwo()
    {
        var checksum = 0;
        foreach (var input in _input)
        {
            foreach (var combination in IterTools<int>.Permutation(input, 2))
            {
                if (combination[0] % combination[1] == 0)
                {
                    checksum += combination[0] / combination[1];
                    break;
                }
            }
        }

        return Convert.ToString(checksum);
    }
}